client/components/Applications/ApplicationsList.jsx

```diff
@@ -8,39 +8,86 @@ export default class ApplicationsList extends Component {
       React.PropTypes.object,
       React.PropTypes.array
     ]).isRequired,
+    groups: React.PropTypes.oneOfType([
+      React.PropTypes.object,
+      React.PropTypes.array
+    ]).isRequired,
     loading: React.PropTypes.bool.isRequired
   }
 
   shouldComponentUpdate(nextProps) {
-    return nextProps.applications !== this.props.applications;
+    return (nextProps.applications !== this.props.applications) ||
+           (nextProps.groups !== this.props.groups);
   }
 
   render() {
-    const { applications } = this.props;
+    const { applications, groups } = this.props;
+
+    const unpackApp = (appId) => {
```

Comment:
Can we not move all of this in the API? Eg, an endpoint returning this:

{
   "Ungrouped Applications": [  xxx ],
   "Group 1": [ ... ]
}